.gitbook/3-protocol-devs/1-dep-specs/5-confidential-oracle.md

```diff
@@ -0,0 +1,50 @@
+# Confidential Oracle
```

Comment:
Your documentation is basically good, but I think you can organize contents much well. I saw many duplicated contents across many sections. I'd suggest the entire plot cleanly like below. You don't have to follow the plot I suggest, but I hope you can see the way how to design the storyline more cleanly.

[Plot Suggestion]

> In DEP, data must be verified as explain in the Data Validation specification.
> -> This cannot be done on chain since data is sensitive. That's why we need the off-chain oracle.
> -> Verifying data means that the oracle should be able to see the contents of data. It also means that oracle operators (human) can see the content of sensitive data. 
> -> _(Here, you can put some examples of malicious behavior. For example, if the oracle software stores something important to the disk without sealing it, it can be read by anyone who can access the disk. Or, if the oracle operator can modify the source code of the oracle to print logs of data contents or to send contents to somewhere else, it would cause serious privacy problems.)_
> -> To prevent it, we have designed the DEP oracle with confidential computing.
> -> Confidential computing guarantees that only selected code (binary) can access the data within the secure enclave. Using it, we can protect sensitive data from being leaked by malicious oracle operators while verifying data correctly in decentralized environments.


Then, you can describe the implementation strategy of confidential oracle. As you've done, you can say that we've used SGX and EGo. 

After that, you can describe how we designed the oracle specifically using SGX and EGo. You've already covered it in the `Additional Efforts` section, introducing unique ID. It is good but you can describe more details.
First, you can explain what an oracle key is, and what if the operator modify the oracle source code, in conjunction with the oracle key.
Then, you can introduce what `MsgRegisterOracle` is, by explaining what the remote report and oracle key mean.
Also, you can introduce how the oracle registration message is handled by other oracles. You can explain what the existing oracles validate.

Reply:
Okay, I think that the plot you suggested seems to be good. I'll reorganize the docs and re-request soon.